### ISSUE TITLE
Custom vars not displayed in searchbar in case it matches the object type.

### ISSUE THREAD

**Issue by tbauriedel** [Icinga/icingadb-web#1094]
## Describe the bug
The searchbar has an bug with "short" custom vars, that matches the object type.

It seems "built-in" variables are prioritized inside the SearchBar. Custom vars are display at the bottom. Also I think there is an hardcoded limit of suggestions that will be displayed.

"os" is part of the string "host". So if you try to search for "os" inside of `icingadb/hosts`  it will always show the "built-in" variables first. The custom var will not be shown.

## To Reproduce
1. Create a custom var "os"
2. Try to use it inside of `icingadb/hosts`
3. You wont find it

## Expected behavior
The custom var should be displayed when it is matching 100%.

## Your Environment
Include as many relevant details about the environment you experienced the problem in.

* Icinga DB Web version (System - About):  1.1.3
* Icinga Web 2 version (System - About): 2.12.1
* icinga/icinga-php-library: 0.14.1
* icinga/icinga-php-thirdparty: 0.12.1

ref/NC/836693

**Comment by nilmerg** [comment]
Indeed, we show only a maximum of 50 suggestions at once. Custom variables are loaded last, so if there are 50 columns, they use up all the available slots.

The original idea with this was that users should bypass this by being more specific. Though, with a name as short as `os` this isn't really an option.

But it also works the other way round: Have plenty of custom variables with `na` in their name and try to find `Host Name`.

What we need here to solve this, is some kind of dynamic limit. At the moment the limit is applied globally on all results. `ObjectSuggestions` should fetch custom variables and collect query columns. If both have more than 25 results, show 25 of each. If one has less than 25, fill up the remaining slots with the other.

**Comment by tbauriedel** [comment]
> What we need here to solve this, is some kind of dynamic limit. At the moment the limit is applied globally on all results. `ObjectSuggestions` should fetch custom variables and collect query columns. If both have more than 25 results, show 25 of each. If one has less than 25, fill up the remaining slots with the other.

Sounds like a good and working solution for me.

`The custom var should be displayed when it is matching 100%. ` 
What do you think about that? in case of a 100% match, it would be nice to have the variable at the top instead of having to scroll down and "search" for it.
